CMakeLists.txt

```diff
@@ -317,7 +317,7 @@ add_subdirectory(third_party)
 if(${BUILD_KUZU})
 add_definitions(-DKUZU_ROOT_DIRECTORY="${PROJECT_SOURCE_DIR}")
 add_definitions(-DKUZU_CMAKE_VERSION="${CMAKE_PROJECT_VERSION}")
-add_definitions(-DKUZU_EXTENSION_VERSION="0.7.0")
```

Comment:
hmm Do we need to upgrade the extension version?

Reply:
I added a virtual function to `TableFuncBindData` which is inherited by some classes in the extensions (e.g. JSONScanBindData) so I probably need to bump the extension version.

That single change (me needing to export `ScanBindData/TableFuncBindData) caused a bunch of dynamic casts in the macos extension test to fail due to the casted targets not being exported which is why I added all those `KUZU_API` changes. I'm not completely sure why they showed up now and not before though.

Reply:
okay. We don't need to bump extension version manually in PRs now. The CI pipeline will handle this.

Reply:
@acquamarin Can you also take a look at the KUZU_API changes in this PR?

Reply:
It probably because the only virtual method before is `copy` which is pure virtual (so the definitions are all in the child classes). The method I added has a default implementation so it actually needs to exported for child classes in the extensions to access.